src/test_helpers.rs

```diff
@@ -40,27 +41,39 @@ impl<'a> Debug for Blob<'a> {
     }
 }
 
+#[derive(Clone)]
 struct Entry {
     name: Vec<u8>,
     data: Vec<u8>,
 }
 
-#[derive(Default)]
+#[derive(Default, Clone)]
 pub struct SimpleStorage {
-    entries: Vec<Entry>,
+    entries: Arc<RwLock<Vec<Entry>>>,
 }
 
 impl SimpleStorage {
     pub fn new() -> SimpleStorage {
-        SimpleStorage { entries: vec![] }
+        SimpleStorage {
+            entries: Arc::new(RwLock::new(vec![])),
+        }
     }
 
-    pub fn has_chunk(&self, name: &[u8]) -> bool {
-        self.entries.iter().any(|entry| entry.name == name)
+    pub async fn has_chunk(&self, name: &[u8]) -> Result<bool, SelfEncryptionError> {
+        Ok(self
+            .entries
+            .read()
+            .map_err(|_| SelfEncryptionError::Poison)?
```

Comment:
better at least log the error?

Reply:
This error happens for test only and there is only one possible error. So I think we should be okay here.